### PR TITLE
Add support for per-file preprocessing

### PR DESCRIPTION
**Pull request by bclinkinbeard**
PR for #3 for initial review

I made the preprocess function the final argument, which avoids breaking the API, but if you'd rather keep the callback as the last arg I can change that. I also couldn't get a "this" ref, so I just stuck it on the `Parser` object directly. The only other difference from what you suggested is that the path is passed into the function as the first arg. We need that info for our purposes, and it is reminiscent of a Browserify transform.

Thoughts?

### CODE REVIEW

File: less.js [hunk 1]
```diff
@@ -5,17 +5,24 @@ var Parser = require("./index")
 
 module.exports = less
 
-function less(location, callback) {
-    var parser = new Parser({
+function less(location, callback, preprocess) {
```
Comment (Raynos) [inline less.js]: could we change this to `less(location, options, callback)`

Then we can do

```
if (typeof options === 'function') {
  callback = options
  options = {}
}
```

This means we'll support both `less(location, callback)` and `less(location, options, callback)`

This is basically a new optional middle argument

**Comment by Raynos** [comment]
passing (src, pathname) is a good idea

**Comment by bclinkinbeard** [comment]
Yep, I will make options the optional second param. 

The signature is currently (pathname, src), do you want me to change the order there too?

**Comment by Raynos** [comment]
@bclinkinbeard i dont mind about the preprocess signature order

**Comment by bclinkinbeard** [comment]
Updated

**Comment by bclinkinbeard** [comment]
FWIW, I have atomify-css working to process assets using these changes. disgonbgud

**Comment by Raynos** [comment]
Released as v0.4.0

**Comment by Raynos** [comment]
Added you as a collaborator on github

**Comment by bclinkinbeard** [comment]
Sweet, thanks!

**Pull request merged**